samples/cellular/modem_shell/src/location/location_details.c

```diff
@@ -0,0 +1,706 @@
+
+	/* Strip out the quotes from operator name */
+	len = strlen(xmonitor_resp.full_name_str);
+	if (len > 2 && len <= OP_FULL_NAME_STR_MAX_LEN) {
+		strncpy(tmp_str, xmonitor_resp.full_name_str + 1, len - 2);
+	}
+	if (!cJSON_AddStringToObjectCS(modem_details_obj, "operator_full_name", tmp_str)) {
+		mosh_error("No memory create json obj operator_full_name");
+		goto cleanup;
+	}
+
+	/* Strip out the quotes from plmn str */
+	memset(tmp_str, 0, sizeof(tmp_str));
+	len = strlen(xmonitor_resp.plmn_str);
+	if (len > 2 && len <= OP_PLMN_STR_MAX_LEN) {
+		strncpy(tmp_str, xmonitor_resp.plmn_str + 1, len - 2);
+	}
+	if (!cJSON_AddStringToObjectCS(modem_details_obj, "plmn", tmp_str)) {
+		mosh_error("No memory create json obj plmn");
+		goto cleanup;
+	}
+
+	if (json_add_num_cs(modem_details_obj, "cell_id", xmonitor_resp.cell_id) ||
+	    json_add_num_cs(modem_details_obj, "pci", xmonitor_resp.pci) ||
+	    json_add_num_cs(modem_details_obj, "band", xmonitor_resp.band) ||
+	    json_add_num_cs(modem_details_obj, "tac", xmonitor_resp.tac) ||
+	    json_add_num_cs(modem_details_obj, "rsrp_dbm", RSRP_IDX_TO_DBM(xmonitor_resp.rsrp)) ||
+	    json_add_num_cs(modem_details_obj, "snr_db",
+			    xmonitor_resp.snr - LINK_SNR_OFFSET_VALUE)) {
+		mosh_error("No memory create modem details json obj");
+		goto cleanup;
+	}
+
+	err = lte_lc_system_mode_get(&sysmode, NULL);
+	if (err) {
+		mosh_warn("lte_lc_system_mode_get failed with err %d", err);
+	} else {
+		if (!cJSON_AddStringToObjectCS(modem_details_obj, "sysmode",
+					       link_shell_sysmode_to_string(sysmode, tmp_str))) {
+			mosh_error("No memory create json obj sysmode");
+			goto cleanup;
+		}
+	}
+
+	if (!cJSON_AddItemToObject(root_obj, "modem_details", modem_details_obj)) {
+		goto cleanup;
+	}
+
+	return 0;
+
+cleanup:
+	/* No need to delete others as they were added to here */
+	cJSON_Delete(modem_details_obj);
+
+	return -ENOMEM;
+}
+
+static int location_details_location_data_encode(
+	const struct location_details_mosh *details_mosh,
+	cJSON *root_obj)
+{
+	cJSON *location_data_obj = NULL;
+	cJSON *method_details_obj = NULL;
+	const struct location_event_data *loc_evt_data = &details_mosh->event_data;
+	const struct location_data_details *details;
+	int err = -ENOMEM;
+
+	details = location_details_get(loc_evt_data);
+	if (details == NULL) {
+		goto cleanup;
+	}
+
+	location_data_obj = cJSON_CreateObject();
+	if (location_data_obj == NULL) {
+		mosh_error("No memory create json obj for details location_data_obj");
+		goto cleanup;
+	}
+
+	if (!cJSON_AddStringToObjectCS(location_data_obj, "method",
+				       location_method_str(loc_evt_data->method))) {
+		mosh_error("No memory create json obj method");
+		goto cleanup;
+	}
+	/* TODO: Maybe we should have event_id or error/timeout visible somewhere?
+	 *       And maybe even fallback info?
+	 */
+	if (!cJSON_AddNumberToObjectCS(location_data_obj, "elapsed_time_method_sec",
+				       (double)details->elapsed_time_method / 1000)) {
```

Comment:
Do we want to show error/timeout/fallback, i.e., event id in some way?
Currently you know failure from the root object's `"data"` field. as the time represented there is -5 and -1 for timeout and failure. Also, `location_data` doesn't have `position` element.

Reply:
Added `event` field into `location_data`.